### PR TITLE
[CURA-11100] scattered seam

### PR DESCRIPTION
**Pull request by wawanbreton**
* Corner angle calculation algorithm was rewritten to better handle differences in resolution between models.
* Adjustment of  start point scores calculation to make the seam consistant and expectable as much as possible

### CODE REVIEW

File: include/PathOrderOptimizer.h [hunk 1]
```diff
@@ -636,18 +636,35 @@ class PathOrderOptimizer
             return vert;
         }
 
+        //Precompute segments lengths because we are going to need them multiple times
+        std::vector<coord_t> segments_sizes(path.converted->size());
+        coord_t total_length = 0;
+        for(const auto& [i, here]: **path.converted | ranges::views::enumerate)
+        {
+            const Point &next = (*path.converted)[(i + 1) % path.converted->size()];
+            coord_t segment_size = vSize(next - here);
```
Comment (casperlamboo) [inline include/PathOrderOptimizer.h:645]:
```suggestion
            const coord_t segment_size = vSize(next - here);
```

File: include/PathOrderOptimizer.h [hunk 2]
```diff
@@ -636,18 +636,35 @@ class PathOrderOptimizer
...
+            segments_sizes[i] = segment_size;
+            total_length += segment_size;
+        }
+
         size_t best_i;
         float best_score = std::numeric_limits<float>::infinity();
         for (const auto& [i, here] : **path.converted | ranges::views::enumerate)
```
Comment (casperlamboo) [inline include/PathOrderOptimizer.h:652]: Rather then performing the if-check on line below for every iteration, you could also skip the last element using the ranges `drop_last` utility
```suggestion
        for (const auto& [i, here] : **path.converted | ranges::views::enumerate | ranges::views::drop_last(/*is_path_closed()*/true ? 1 : 0))
```

**Review by casperlamboo — COMMENTED** [review]
Had some very minor remarks these were mostly personal preference remarks as I had not really any code-remarks

File: include/PathOrderOptimizer.h (hunk 2 above)
Comment (wawanbreton) [reply include/PathOrderOptimizer.h:652]: This is very nice, I was looking for a clean way to do this but I'm not familiar enough with `std::ranges`

**Review by casperlamboo — APPROVED** [review]

**Pull request merged**